sycl/include/CL/sycl/detail/buffer_impl.hpp

```diff
@@ -168,6 +182,8 @@ class __SYCL_EXPORT buffer_impl final : public SYCLMemObjT {
 
   ~buffer_impl() {
     try {
+      if (BaseT::MInteropContext && !BaseT::MOwnNativeHandle)
+        BaseT::updateInteropMemory();
```

Comment:
What is this doing? Maybe add comment if still need something special?

Reply:
Buffer can be accessed not only in the context where it was created but in another contexts as well. In this case scheduler creates separate instance of pi_mem (which has it's own memory allocation) and copies buffer contents from one pi_mem to another. BaseT::updateInteropMemory() copies the latest buffer data back to the original memory allocation provided by user. 
Or in simple case buffer can be accessed and changed on host using host accessor. Then BaseT::updateInteropMemory() will copy from host to original memory allocation.

Reply:
Added comments.